ecmascript/transforms/module/src/amd.rs

```diff
@@ -300,10 +305,13 @@ impl Fold for Amd {
 
                         // export { foo } from 'foo';
                         ModuleDecl::ExportNamed(export) => {
+                            let mut scope_ref_mut = self.scope.borrow_mut();
+                            let scope = &mut *scope_ref_mut;
                             let imported = export.src.clone().map(|src| {
-                                self.scope
-                                    .import_to_export(&src, !export.specifiers.is_empty())
+                                scope.import_to_export(&src, !export.specifiers.is_empty())
                             });
+                            drop(scope);
```

Comment:
Why is this required?

Reply:
Without it I get this error:

```
error[E0502]: cannot borrow `*self` as mutable because it is also borrowed as immutable
   --> ecmascript/transforms/module/src/amd.rs:361:54
    |
308 |                             let mut scope_ref_mut = self.scope.borrow_mut();
    |                                                     ---------- immutable borrow occurs here
...
361 |                                     None => Box::new(Expr::Ident(orig.clone()).fold_with(self)),
    |                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
...
409 |                         }
    |                         - immutable borrow might be used here, when `scope_ref_mut` is dropped and runs the destructor for type `RefMut<'_, Scope>`
```

As far as I understand, `scope` and `scope_ref_mut` constitute an immutable borrow of `self` which will not be dropped till near the bottom of this function.  This conflicts with mutable borrows made by `fold_with(self)`.  But if I drop `scope` and `scope_ref_mut` earlier, there is no conflict.